### PR TITLE
[PORT] Vault bank machine checks for vault access before alerting

### PR DESCRIPTION
**Pull request by warface1234455**
# About pr

- ports: https://github.com/tgstation/tgstation/pull/68832

# Document the changes in your pull request
Vault bank now checks for access for alerting

# Wiki Documentation

Vault bank now checks for access for alerting

# Changelog

:cl:  [distributivgesetz](https://github.com/distributivgesetz) ported by warface1234455
rscadd: Vault bank now check for access for alerting
/:cl:

### CODE REVIEW

File: code/game/machinery/bank_machine.dm [hunk 1]
```diff
@@ -79,21 +80,30 @@
...
-/obj/machinery/computer/bank_machine/ui_act(action, params)
+/obj/machinery/computer/bank_machine/ui_act(action, params, datum/tgui/ui)
 	if(..())
 		return
 
 	switch(action)
 		if("siphon")
 			say("Siphon of station credits has begun!")
-			siphoning = TRUE
+			start_siphon(ui.user)
```
Comment (redmoogle) [inline code/game/machinery/bank_machine.dm]:
```suggestion
			if(iscarbon(usr))
				var/mob/living/carbon/C = usr
				start_siphon(C)
```

Comment (redmoogle) [reply code/game/machinery/bank_machine.dm]: Other code uses this, likely a better choice as I see no example of yours besides in slime people

**Review by redmoogle — CHANGES_REQUESTED** [review]

**Comment by JamieD1** [comment]
Noone should be taking money anyway, so alerting on everyone is fine.

**Comment by ShoziOffical** [comment]
> Noone should be taking money anyway, so alerting on everyone is fine.

It still alert people tho, radio scream for unauthorized people and no scream for authorized people

**Comment by JamieD1** [comment]
> > Noone should be taking money anyway, so alerting on everyone is fine.
> 
> It still alert people tho, radio scream for unauthorized people and no scream for authorized people

I want the captain to be screamed at, he shouldn't be taking money from the vault

**Pull request closed without merging**